internal/internal_nexus_task_handler.go

```diff
@@ -393,6 +396,9 @@ func (h *nexusTaskHandler) newNexusOperationContext(response *workflowservice.Po
 		TaskQueue:      h.taskQueueName,
 		MetricsHandler: metricsHandler,
 		Log:            logger,
+		ResolveWorkflowName: func(a any) (string, error) {
```

Comment:
This would be clearer as a method instead of a lambda property IMO (if you need to store the registry on the context on a non-exported field, no prob)